config/initializers/console.rb

```diff
@@ -10,4 +10,35 @@ def find(id)
       raise "Don't know how to resolve this ¯\\_(ツ)_/¯. Please provide a valid email or Global ID."
     end
   end
+
+  def hard_delete_invoice(id)
+    invoice = Invoice.find(id)
+    puts "Going to hard delete invoice from org `#{invoice.organization.name}` (id: #{invoice.id})" # rubocop:disable Rails/Output
+
+    puts "Press any key to confirm deletion or CTRL+C to cancel." # rubocop:disable Rails/Output
+    c = $stdin.getch
+
+    if c == "\u0003"
+      puts "Deletion cancelled." # rubocop:disable Rails/Output
+      return invoice
+    end
+
+    puts "Deleting invoice #{invoice.id}..." # rubocop:disable Rails/Output
+    ActiveRecord::Base.transaction do
+      invoice.invoice_subscriptions.destroy_all
+      invoice.credit_notes.destroy_all
+      invoice.fees.each { |f| f.true_up_fee&.destroy! }
+      invoice.fees.destroy_all
+      invoice.taxes.destroy_all
+      invoice.credits.destroy_all
+      invoice.destroy!
+    end
+
+    begin
+      invoice.reload
+      puts "Invoice #{id} could not be deleted. Please try again." # rubocop:disable Rails/Output
+    rescue ActiveRecord::RecordNotFound
+      puts "Invoice #{id} has been successfully deleted." # rubocop:disable Rails/Output
```

Comment:
I think those two `puts` commands should switch positions :D

Reply:
In this case we use `.reload` to see if the record still exists in the db. When you get an not found error, it's a confirmation that the model was deleted. If you're able to reload it, it means it's still there so the deletion failed.
But I agree it's very confusing!